### ISSUE TITLE
Wifi radio control not working on Android 10/11

### ISSUE THREAD

**Issue by balporig** [henrichg/PhoneProfilesPlus#84]
Wondering if this is working on the F-Droid version with the special wifi permission granted. Tested on 2 different non-rooted phones one with Andorid 10 and another with 11 and it does seem there is no control over the wifi state.

**Comment by henrichg** [comment]
Version of PPP is the same in all application stores.
Please what you mean: "wifi radio control"? Switch Wi-Fi on/off?
1. Check in PPP Settings/Permissions/Wi-Fi control special access.
2. For Xiaomi devices with MIUI:  PPP Settings/Permissions/Wi-Fi and Bluetooth confirm dialogs. 
3
And please, instal latst PPP version (5.3.1).

**Comment by balporig** [comment]
Yes, I mean Switch Wi-Fi on/off. The latest build is used in both places (`5.3.1`). Also, as mentioned Settings/Permissions/Wi-Fi control special access has been granted. In that case is a bug. Some more contexts for the hardware/software BOM used for testing:

- Samsung Galaxy S9 with Android 10 and OneUI 2.5
- Xiaomi Mi A2 with CalyxOS 2.14.0 - Android 11 which is based on LinegageOS (Android AOSP)

**Comment by henrichg** [comment]
With manual profile activation not working? 

I own Galaxy S10 (Android 11) and S21 (Android 12) and in both devices is not problem with switch wi-fi.
Maybe problem is with airplane mode. When is enabled, wi-fi switch not workig and this is OK. Must not working in PPP.

Send me PPP dat. For this use in Editor "Support"/"E-mail settings to author".

**Comment by henrichg** [comment]
For airpane mode. Read this: https://github.com/henrichg/PhoneProfilesPlus/blob/devel/docs/airplane_mode_radios_config.md

**Comment by balporig** [comment]
> For airpane mode. Read this: https://github.com/henrichg/PhoneProfilesPlus/blob/devel/docs/airplane_mode_radios_config.md

The devices were in airplane mode. When disabled, toggling Wifi works, just tested it! The guide above lets you exclude radios from being toggled by airplane mode. This is not the issue

> Maybe problem is with airplane mode. When is enabled, wi-fi switch not workig and this is OK. Must not working in PPP.

That's it, in that case this is a feature request for PPP to support toggling Wifi when Airplane mode is enabled, updated the issue title

**Comment by henrichg** [comment]
Sorry, but is not possible to enable Wi-Fi when airplane mode is enabled. Android SDK API do not allow this. 
WifiManager,setWifiEnabled(true) do nothing, when airplane mode is on. :-(

**Comment by balporig** [comment]
Interesting, probably it's not exposed by the SDK API but works done manually from the UI.
How about if it's implemented like this: you want to toggle Wifi and you're in airplane mode => Disable airplane mode => Togggle Wifi => Enable Airplane mode. In conjuction with https://github.com/henrichg/PhoneProfilesPlus/blob/devel/docs/airplane_mode_radios_config.md should be doable with the Andoid SDK API, right?

**Comment by henrichg** [comment]
In PPP, airplane mode is set as first. After 2,5 secods is set Wi-Fi. You can configure airplane mode radios (as is writen in https://github.com/henrichg/PhoneProfilesPlus/blob/devel/docs/airplane_mode_radios_config.md): all, except wi-fi. And then you can switch wi-fi as you want, when airplane mode is enabled.

**Comment by balporig** [comment]
I'm aware of this, the logical workflow above was an example of how this can probably be implemented in an automated manner as part of the PPP's codebase. In ideal case PPP should be intelligent enough to be able to handle what to do in that scenario as the whole point is not to be done manually. There are 2 approaches, either it can be implemented with a workaround or there should be a disclaimer this is not supported, and the user should use Automate/Tasker workflow to implement it. But isn't the whole purpose of PPP not to use one of those apps for profile management?

**Comment by henrichg** [comment]
Hm, hm. I'm check airplane mode + wi-fi on in Macrodroid and this is possible and working. Hm, hm.
Then I must searching solution for this. Hm hm. Again Macrodorid. :-)
I will looking for solution.

**Comment by balporig** [comment]
Thank you :)

**Comment by henrichg** [comment]
Good good. Found solution. ;-)
1. In PPP airplane mode ON + Blueottoh ON working as in Macrodroid. Only must be configured in PPP in profile both Airplane mode ON and Bluetooth ON;-)
2. For Wi-Fi is in Macrodroid used command "svc wifi enable", instead of WifiManager.setWifiEnabled(). Is possible to implement special version of "Airplane mode ON" with Wi-Fi ON, but I must test, if this command working without root. By Macrodroid is not possible to configure "keep Wi-Fi enabled" when used is Goofle assistant version. Only in root version of airplane mode.

**Comment by balporig** [comment]
Nice! Sounds like something that can work

**Comment by henrichg** [comment]
OK, implemented amd by my tests working good. It is on this beta: https://drive.google.com/file/d/1CR4y2u3V8PCwQK9sw1P88JrlpSdA2Ou4/view?usp=drivesdk

If you want, install it.
1. Do backup of PPP data: Editor menu/"Backup/Restore"/"Backup settings".
2. Install beta

4. When in beta are bugs, problems (send me also crash, when will be generated):
5. Unistall PPP.
6. Install PPP from application store.
7. Do restore of PPP data: Editor menu/"Backup/Restore"/"Restore settings".

**Comment by balporig** [comment]
I was not able to test the new options as both of my devices are not rooted, but have some good news. Got switching Wifi on/off in airplane mode working without root using the regular ON/OFF (`WifiManager.setWifiEnabled(true)`):

1. Set PPP as default assistant for working Airplane mode switching
2. The wifi special access usually is always granted and no manual intervention needed
3. Grant G1 permission [PhoneProfilesPlus/grant_g1_permission.md at master · henrichg/PhoneProfilesPlus](https://github.com/henrichg/PhoneProfilesPlus/blob/master/docs/grant_g1_permission.md)
4. Excluded wifi from Airplane toggle [PhoneProfilesPlus/airplane_mode_radios_config.md at devel · henrichg/PhoneProfilesPlus](https://github.com/henrichg/PhoneProfilesPlus/blob/devel/docs/airplane_mode_radios_config.md)
5. Reboot the device

And it works. Tested it on both devices. The only difference between my previous attempts and now is step `5`

Also wanted to say I'm really grateful for all the effort you've put into PPP and keeping it opensource. It shows how much time and care was invested. Implementing profile management in any other mobile automation (Automate/Tasker/Macrodroid etc.) is a nightmare :)
